platform/critical.h

```diff
@@ -18,342 +15,5 @@
- *     done = false
- *     value = *p // This fetch operation need not be atomic.
- *     while not done {
- *         done = atomic_cas(p, &value, value + a) // *value gets updated automatically until success
- *     }
- *     return value + a
- * }
- */
-bool core_util_atomic_cas_ptr(void **ptr, void **expectedCurrentValue, void *desiredValue);
-
-/**
- * Atomic increment.
- * @param  valuePtr Target memory location being incremented.
- * @param  delta    The amount being incremented.
- * @return          The new incremented value.
- */
-uint8_t core_util_atomic_incr_u8(uint8_t *valuePtr, uint8_t delta);
-
-/**
- * Atomic increment.
- * @param  valuePtr Target memory location being incremented.
- * @param  delta    The amount being incremented.
- * @return          The new incremented value.
- */
-uint16_t core_util_atomic_incr_u16(uint16_t *valuePtr, uint16_t delta);
-
-/**
- * Atomic increment.
- * @param  valuePtr Target memory location being incremented.
- * @param  delta    The amount being incremented.
- * @return          The new incremented value.
- */
-uint32_t core_util_atomic_incr_u32(uint32_t *valuePtr, uint32_t delta);
-
-/**
- * Atomic increment.
- * @param  valuePtr Target memory location being incremented.
- * @param  delta    The amount being incremented in bytes.
- * @return          The new incremented value.
- *
- * @note The type of the pointer argument is not taken into account
- *       and the pointer is incremented by bytes.
- */
-void *core_util_atomic_incr_ptr(void **valuePtr, ptrdiff_t delta);
-
-/**
- * Atomic decrement.
- * @param  valuePtr Target memory location being decremented.
- * @param  delta    The amount being decremented.
- * @return          The new decremented value.
- */
-uint8_t core_util_atomic_decr_u8(uint8_t *valuePtr, uint8_t delta);
-
-/**
- * Atomic decrement.
- * @param  valuePtr Target memory location being decremented.
- * @param  delta    The amount being decremented.
- * @return          The new decremented value.
- */
-uint16_t core_util_atomic_decr_u16(uint16_t *valuePtr, uint16_t delta);
-
-/**
- * Atomic decrement.
- * @param  valuePtr Target memory location being decremented.
- * @param  delta    The amount being decremented.
- * @return          The new decremented value.
- */
-uint32_t core_util_atomic_decr_u32(uint32_t *valuePtr, uint32_t delta);
-
-/**
- * Atomic decrement.
- * @param  valuePtr Target memory location being decremented.
- * @param  delta    The amount being decremented in bytes.
- * @return          The new decremented value.
- *
- * @note The type of the pointer argument is not taken into account
- *       and the pointer is decremented by bytes
- */
-void *core_util_atomic_decr_ptr(void **valuePtr, ptrdiff_t delta);
-
-#ifdef __cplusplus
-} // extern "C"
-#endif
-
-
-#endif // __MBED_UTIL_CRITICAL_H__
-
-/** @}*/
+#warning critical.h has been replaced by mbed_critical.h, please update to mbed_critical.h [since mbed-os-5.3]
```

Comment:
The included file has the protection against multiple inclusion, the only downside is multiple warning messages which shouldn't be an issue.